client/app/services/service-details/service-details-ansible.component.js

```diff
@@ -106,4 +112,15 @@ function ComponentController (ModalService, ServicesState, lodash) {
   function elapsed (finish, start) {
     return Math.abs(new Date(finish) - new Date(start)) / 1000
   }
+
+  function processCredential (item) {
+    item.type = item.type.substring(item.type.lastIndexOf('::') + 2, item.type.lastIndexOf('Credential'))
+    return item
+  }
+
+  function normalizeDialogKeys (key) {
+    const normalizedKey = key.replace(/dialog_/i, '')
+
+    return normalizedKey === 'credential' ? 'credential_id' : normalizedKey
```

Comment:
To include other credential type, should it be something like
```
  return normalizedKey.contains('credential') ? normalizedKey+'_id' : normalizedKey
```

Reply:
@bzwei if at dialog has a credential override of `dialog_cloud_credential_id` the change you are proposing would result in a normalized key of `cloud_credential_id_id`

This conditional is to catch the edge case of `dialog_credential` which would be normalized to `credential` which would not override the desired machine credential which is called `credential_id`

Reply:
>  Since you only need to look for one and only credential, you do not need to use a map.

unclear watcha mean by this ☝️ 😕

Reply:
I thought you use a map from `credential` to value. In this map it can be only one pair. why bother?

Reply:
In the case of `provision ` and `dialog` present, we map through each `dialog` keys, normalize them and assign them to the config_info object. Without this we miss all overrides 😬